### ISSUE TITLE
Socket forceably closed exceptions should not show up for default logging experience

### ISSUE THREAD

**Issue by danroth27** [aspnet/KestrelHttpServer#2254]
When testing an application using a browser the browser will often leave the socket open for a while after the request has completed. If you close the browser while the socket is still open you see the following log message with our default templates in development:

```
fail: Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets[0]
      Unexpected exception in SocketConnection.StartAsync.
System.Net.Sockets.SocketException (10054): An existing connection was forcibly closed by the remote host
   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName)
   at System.Net.Sockets.Socket.Shutdown(SocketShutdown how)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Internal.SocketConnection.<DoSend>d__16.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Internal.SocketConnection.<StartAsync>d__14.MoveNext()
```

The fact that the client forcibly closed the connection should not be an error on the server. Clients do this all the time. I think a debug level message would be more appropriate.

**Comment by Tratcher** [comment]
Clarification: The closure of idle sockets should not be logged as an error.

Logistically this may present a challenge as the transport layer does not know when a socket is idle. Perhaps the transport shouldn't log this error at all, it should throw it up to the protocol layer so they can determine if the current state is idle and adjust the log level accordingly.

**Comment by muratg** [comment]
@davidfowl @halter73 any way to do this in a transport agnostic way? Do we want to do this anytime soon?

(Backlogging for now, but if it's important enough, we'll bring hit back.)

**Comment by seanmars** [comment]
When i ctrl+c to close the server it will get exception about "Unexpected exception in SocketConnection.StartAsync"

```
Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets|Unexpected exception in SocketConnection.StartAsync
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'MemoryPoolBlock'.
   at System.Buffers.ThrowHelper.ThrowObjectDisposedException(ExceptionArgument argument)
   at System.Buffers.MemoryPoolBlock.Dispose()
   at System.IO.Pipelines.BufferSegment.ResetMemory()
   at System.IO.Pipelines.Pipe.CompletePipe()
   at System.IO.Pipelines.Pipe.CompleteWriter(Exception exception)
   at System.IO.Pipelines.Pipe.DefaultPipeWriter.Complete(Exception exception)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Internal.SocketConnection.DoReceive()
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Internal.SocketConnection.StartAsync(IConnectionDispatcher connectionDispatcher)
```

**Comment by halter73** [comment]
The error log originally reported by @danroth27 was fixed by #2547

```
fail: Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets[0]
      Unexpected exception in SocketConnection.StartAsync.
System.Net.Sockets.SocketException (10054): An existing connection was forcibly closed by the remote host
   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName)
   at System.Net.Sockets.Socket.Shutdown(SocketShutdown how)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Internal.SocketConnection.<DoSend>d__16.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Internal.SocketConnection.<StartAsync>d__14.MoveNext()
```
